backend/src/main/java/asegroup1/api/services/landregistry/LandRegistryServiceImpl.java

```diff
@@ -56,11 +61,11 @@ public LandRegistryServiceImpl() throws IOException {
             );
         }
 
-        return addressList;
+        return getPositionForAddresses(addressList);
     }
 
-    public List<AddressWithTransaction> getTransactionsByPostCode(String postcode) throws IOException, UnirestException, ParseException {
-        List<AddressWithTransaction> transactionsList = new LinkedList<>();
+    public List<Address> getTransactionsForPostCode(String postcode) throws IOException, UnirestException, ParseException {
```

Comment:
Why is this changed to be more vague? By making getPositionForAddresses generic you can keep the original return type.